ocaml-lsp-server/src/compl.ml

```diff
@@ -87,7 +88,14 @@ let prefix_of_position ~short_path source position =
       | Some pos -> pos + 1
     in
     let len = from - pos + 1 in
-    String.sub text ~pos ~len
+    let reconstructed_prefix = String.sub text ~pos ~len in
+    (* if we reconstructed [~f:ignore], we should take only [ignore], so: *)
+    if not (String.is_prefix reconstructed_prefix ~prefix:"~") then
```

Comment:
Without looking at this very deeply, why doesn't the problem exist for optional arguments?

Reply:
Excellent remark. Fixed handling for optional args as well.